state-chain/pallets/cf-elections/src/electoral_systems/tests/solana_vault_swap_accounts.rs

```diff
@@ -0,0 +1,309 @@
+		},
+		initiate_vault_swap_hook_not_called(_pre, _post) {
+			assert_eq!(INITIATE_VAULT_SWAP_CALLED.with(|hook_called| hook_called.get()), 0, "Hook should have been called once so far!");
+		},
+		initiate_vault_swap_hook_called_twice(_pre, _post) {
+			assert_eq!(INITIATE_VAULT_SWAP_CALLED.with(|hook_called| hook_called.get()), 2, "Hook should have been called once so far!");
+		},
+		initiate_vault_swap_hook_called_four_times(_pre,_post) {
+			assert_eq!(INITIATE_VAULT_SWAP_CALLED.with(|hook_called| hook_called.get()), 4, "Hook should have been called once so far!");
+		},
+		initiate_vault_swap_hook_called_15_times(_pre, _post) {
+			assert_eq!(INITIATE_VAULT_SWAP_CALLED.with(|hook_called| hook_called.get()), 15, "Hook should have been called once so far!");
+		},
+		close_accounts_hook_not_called(_pre, _post) {
+			assert_eq!(CLOSE_ACCOUNTS_CALLED.with(|hook_called| hook_called.get()),	0, "Hook should not have been called!");
+		},
+		close_accounts_hook_called_once(_pre, _post) {
+			assert_eq!(CLOSE_ACCOUNTS_CALLED.with(|hook_called| hook_called.get()),	1, "Hook should not have been called!");
+		},
+		close_accounts_hook_called_twice(_pre, _post) {
+			assert_eq!(CLOSE_ACCOUNTS_CALLED.with(|hook_called| hook_called.get()),	2, "Hook should not have been called!");
+		},
+		get_sol_nonces_hook_not_called(_pre, _post) {
+			assert_eq!(GET_NUMBER_OF_SOL_NONCES_CALLED.with(|hook_called| hook_called.get()), 0, "Hook should not have been called!");
+		},
+		get_sol_nonces_hook_called_once(_pre, _post) {
+			assert_eq!(GET_NUMBER_OF_SOL_NONCES_CALLED.with(|hook_called| hook_called.get()), 1, "Hook should not have been called!");
+		},
+		get_sol_nonces_hook_called_twice(_pre, _post) {
+			assert_eq!(GET_NUMBER_OF_SOL_NONCES_CALLED.with(|hook_called| hook_called.get()), 2, "Hook should not have been called!");
+		},
+	}
+}
+
+#[test]
+fn on_finalize_accounts_limit_reached() {
+	TestSetup::default()
+		.with_unsynchronised_state(0)
+		.build()
+		.test_on_finalize(
+			&0u32,
+			|_| {
+				assert_eq!(
+					MockHook::close_accounts_called(),
+					0,
+					"Hook should not have been called!"
+				);
+				assert_eq!(MockHook::init_swap_called(), 0, "Hook should not have been called!");
+				assert_eq!(
+					MockHook::get_number_of_available_sol_nonce_accounts_called(),
+					0,
+					"Hook should not have been called!"
+				);
+			},
+			vec![
+				Check::<MinimalVaultSwapAccounts>::only_one_election(),
+				Check::<MinimalVaultSwapAccounts>::initiate_vault_swap_hook_not_called(),
+				Check::<MinimalVaultSwapAccounts>::close_accounts_hook_not_called(),
+				Check::<MinimalVaultSwapAccounts>::get_sol_nonces_hook_not_called(),
+			],
+		)
+		.expect_consensus(
+			generate_votes_n_to_m_accounts(0, 15),
+			Some(SolanaVaultSwapsVote {
+				new_accounts: (0..15u64).map(|i| (i, ())).collect::<BTreeSet<_>>(),
+				confirm_closed_accounts: BTreeSet::from([CLOSED_ACCOUNT_1, CLOSED_ACCOUNT_2]),
+			}),
```

Comment:
The construction here is a little strange, we are duplicating code here and then inside generate_votes_n_to_m_accounts. Makes it hard to see why this expectation is actually true just from looking at the test. It's only true because it happens to exactly match what's in `generate_votes_n_to_m_accounts` but this could be explicit at the level of the test

Reply:
This test is actually about testing the on_finalize(). It is not testing consensus and so It is not so important if consensus is correct. We just need to provide some consensus value which is then used to test on_finalize so I think it is fine.

Reply:
`.expect_consensus(` means it is testing consensus. If the check_consensus result changes then it impacts the on_finalize so it's no less important

Reply:
I meant that the test is not suppose to test the consensus so it is not important. I was using the ```expect_consensus``` function to update the consensus. I changed to using the dedicated function to update the consensus (```force_consensus_update```) so that this fact is clear.